governance/API-Ref-Guidelines-Update.md

```diff
@@ -0,0 +1,508 @@
+- [Excluding code from the documentation](#excluding-code-from-the-documentation)
+- [Exceptions](#exceptions)
+- [As an SDE, can I build the documentation to see the final result?](#as-an-sde-can-i-build-the-documentation-to-see-the-final-result)
+
+## Scope
+
+This page describes how to comment O3DE's C++ code to produce consistent, high-quality documentation for O3DE’s public-facing C++ API Reference Guide. The intended audiences of _this page_ are O3DE SDE's and programming writers.
+
+We list the minimum code elements to document, describe how to add the documentation, and provide examples of documenting different types of elements. Descriptions of how to build and publish the API Reference Guide are out of scope of this document.
+
+## Who is the customer?
+
+An API Reference consists of two major parts
+
+1. The actual header and cpp files that are shipped as part of O3DE, viewed by SDEs in their IDE as they work on the code.
+2. Generated API Reference Guides, generated by [Doxygen](http://www.stack.nl/~dimitri/doxygen/), which is a standard tool for producing documentation from C++ code.
+
+As of the time of writing this guide, the customer for the API reference are white-glove customers and internal studios using O3DE. Zero of these customers have called out any specific interest in reading generated API Reference guides, and in general, they all prefer to read header files and source code to understand APIs.
+
+Note: We have configured Doxygen to ignore "docs.h" files, which solely contain documentation. Docs for specific APIs will be moved to the corresponding header files, and extensive documentation will be moved to the [O3DE Developer Guide](https://o3de.org/docs/).
+
+## How much documentation should I add?
+
+Doxygen distinguishes between two types of descriptions: brief descriptions and detailed descriptions. With our configuration, a brief description is the first sentence of a Doxygen-formatted code comment (described later). A detailed description is anything that follows the first sentence.
+
+Do not use `@brief` or `/brief`. Our doxygen configuration automatically assumes the first sentence (ending with a period) is the brief description in a comment block or multiline comment.
+
+In general, when deciding what to document and how much, just consider the customer whenever adding documentation.  The customer is primarily an SDE reading the public API headers and trying to quickly figure out what APIs are available, what they do, and how to use them. This means that information that is redundant and provides no extra information is of negative value, as it will clutter the header and make it harder to read. But information that is useful to a reader such as input constraints, explanation of a class's purpose, references to other related structures, explanation of parameters that have idiosyncrasies, and other explanations that help a reader of the header understand how to use the API are all recommended.
```

Comment:
...consider the target audience whenever adding documentation." etc.   SDE, etc, de-amazonify.

Reply:
I'm just trying to get a first pass of this added 😂 I didn't write any of this to start with 😝 I was hoping we can iterate after but I can certainly make a few more changes now